apache-apisix/metadata.csv

```diff
@@ -0,0 +1,12 @@
+metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name
+apisix.request.counter,count,,request,,Number of requests received.,0,apache_apisix,
+apisix.request.latency,gauge&rate,,millisecond,,"The latency added by Apache APISIX, time taken to process the request.",0,apache_apisix,
+apisix.upstream.latency,gauge,,millisecond,,"The latency added by Apache APISIX, time taken since proxying the request to the upstream server till a response is received.",0,apache_apisix,
+apisix.upstream.latency,rate,,millisecond,,"The latency added by Apache APISIX, time taken since proxying the request to the upstream server till a response is received.",0,apache_apisix,
+apisix.apisix.latency,gauge,,millisecond,,"The latency added by Apache APISIX, time taken by APISIX agent to process the request.",0,apache_apisix,
+apisix.apisix.latency,rate,,millisecond,,"The latency added by Apache APISIX, time taken by APISIX agent to process the request.",0,apache_apisix,
+apisix.ingress.size,gauge,,byte,,Request body size in bytes.,0,apache_apisix,
+apisix.ingress.size,rate,,byte,,Request body size in bytes.,0,apache_apisix,
+apisix.egress.size,gauge,,byte,,Response body size in bytes.,0,apache_apisix,
+apisix.egress.size,rate,,byte,,Response body size in bytes.,0,apache_apisix,

```

Comment:
Description seems a bit generic, is it possible to add the differences between these 4 metrics inside the description?

Reply:
sure, @bisakhmondal could you please give some more details descriptions?

Reply:
Hi @coignetp, updated. In dogstatsd, the metric type `timer` is a combination of `gauge` and `rate`. That's why we thought bisecting it would be a good idea, but it seems we can't have duplicates. So just ended up updating it to `histogram` - the closest match. Please share your views here.

Reply:
It didn't work. Seems, we have to put down the primitive ones (rate, gauge). How do we do this? It would be really helpful : )